\section{Introduction}
\normalsize
Outline
\begin{itemize}
    \item OF, OFs, orthologs, paralogs, etc
    \item evolutionary models: DTL vs DLC
    \item HMMs \& advantage of position-independent evolutionary models
    \item Metabolic network expansion -> more? what is it, what are the models, etc.
    \item Metabolic network expansion traditionally used to infer evolution of metabolism. Study objective. Essentially connecting phylogenetic analysis with met. net. exp.
\end{itemize}


\subsection*{A surge in available data}
\addcontentsline{toc}{subsubsection}{A surge in available data}

The advancement of sequencing technologies and the development of state-of-the-art bioinformatics tools have enabled a rapid increase not only in the number of sequenced genomes, but also in the sampled prokaryotic diversity, expanding our view of the tree of life, and especially that of the archaeal branch. This surge in data availability can readily improve the resolution of modern phylogenetic relationships, resolve further their deep evolutionary history, and provide more accurate inference of ancestral gene contents \cite{tahon2021}. The first release of GTDB back in 2019, R89, contained 145,904 genomes organized into 24,706 species clusters, while its latest, R220 release this April, has grown to contain 596,859 genomes organized into 113,104 species clusters \cite{parks2018, parks2020, parks2022, rinke2021}; an increase of more than 300\% in just five years.

To date, several studies have investigated the nature and physiology of LACA, LBCA, and LUCA (Last Universal Common Ancestor) through gene content estimation \cite{williams2017, xavier2021, moody2024, coleman2021}, with datasets of various sizes and microbial physiologies, originating from a number of different databases. Usually these datasets are optimized for tree rooting and take into account taxon sampling evenness, yet it is still unclear how the incorporation of more species or other species belonging to the same higher taxonomic level will affect the analysis.
 
Moreover, GTDB, in its effort to standardize prokaryotic taxonomy, employs rank-normalization for the classification of microorganisms at all taxonomic levels \cite{parks2018}, which leads to periodic reclassification and renaming of taxa in the database. The change in nomenclature and the constant taxonomic revisions create confusion even when referencing taxa from the same database, and hinder the refinement of deep phylogenetic relationships \cite{tahon2021}. Especially in an era of exponential growth in genomic data, there is a pressing need to determine the robustness of standard methodologies and result stability. 