\newpage
\thispagestyle{empty}
\section*{\textbf{Abstract}}

\large 

Metabolic Network Expansion (MNE) and its applications have been used to explore the metabolic potential, symbiotic interactions, and minimal nutritional requirements of extant life. Although this heuristic approach does not account for evolutionary constraints, it has been suggested to display evolution-like features and has been used to make inferences about metabolic evolution. In this study, we hypothesize that the expansion process is predetermined rather than evolution-like, and demonstrate that the flow of chemical information within the network is influenced by the nature, topology, and number of compounds used to initialize the expansion. 

Furthermore, our investigation into the impact of taxon sampling on ancient gene content inference uncovers several often overlooked issues in bioinformatics pipelines: taxonomic unevenness that is independent of taxon abundance, discrepancy of gene clustering algorithms and functionality assignment between the various tools and databases, and missing key information from one of the most widely used databases for metabolic modeling. To reconstruct the metabolic networks of putative ancient microorganisms, we apply phylogenetic constraints to MNE, which provides new insights into the origin and evolution of modern metabolism. Finally, our results showcase that increasing availability of data may reshape our understanding of the physiology of the earliest microorganisms that inhabited Earth.